### DISCUSSION TITLE
Has release happened or not?

### DISCUSSION THREAD

**Discussion by laughedelic** [intuit/auto#1959] [Q&A]
Is there an easy way to run `auto shipit` and know if release (any: latest or canary) has happened or was skipped? Running it with `--dry-run --quiet` doesn't help because canary release will still output some (would-be-released) version.

**Comment by hipstersmoothie** [comment]
The afterShipitHook exposes that information. I'm open to suggestions on how to solve your use case 😄

**Reply by laughedelic** [reply]
Yes, I've tried something like

```json
  "plugins": [
    [
      "exec",
      {
        "afterShipIt": "echo $ARG_0 > /tmp/release.json"
      }
    ],
    "git-tag"
  ],
```

And then read this file like `cat /tmp/release.json | jq -r '.newVersion'`. But it feels very fragile and failed in some cases (I'm not sure now when exactly) and I moved away from it.

Isn't there any more robust way to have some indication of whether release happened or not?

**Reply by laughedelic** [reply]
How about a `--json` CLI flag for the `shipit` command so that all the information about release: skipped or not, version bump, new version, changelog, etc. can be then extracted by other tools, i.e either written to a JSON file and parsed or piped to `jq` to extract a single value. 

I'm not sure about the logs in this case though, it would be nice to keep the logs on stderr.

**Reply by laughedelic** [reply]
Or alternatively a `--report <file>` option that would do that `"afterShipIt": "echo $ARG_0 > <file>"` bit without the need for `exec` plugin? I don't think this JSON contains all information though.

**Reply by hipstersmoothie** [reply]
How about a plugin (in the repo) that does it for you? that way we can have lots of options contained within the plugin without complicating the core API.

**Reply by laughedelic** [reply]
Do you mean a plugin that will report release information to a JSON file? 🤔 Sure, that could work.
